common/src/main/java/com/microsoft/identity/common/internal/controllers/BrokerMsalController.java

```diff
@@ -220,14 +227,21 @@ String hello(final @NonNull IIpcStrategy strategy,
     String hello(final @NonNull IIpcStrategy strategy,
                  final @Nullable String minRequestedVersion,
                  final @NonNull String clientMaxProtocolVersion) throws BaseException {
+        final String methodTag = TAG + ":hello";
 
-        final String cachedProtocolVersion = mHelloCache.tryGetNegotiatedProtocolVersion(
-                minRequestedVersion, clientMaxProtocolVersion);
+        final String cachedProtocolVersion = tryGetNegotiatedProtocolVersionFromHelloCache(
+                minRequestedVersion,
+                clientMaxProtocolVersion
+        );
 
-        if (!StringUtil.isEmpty(cachedProtocolVersion)) {
+        if (cachedProtocolVersion != null) {
```

Comment:
>if (cachedProtocolVersion != null) { [](http://example.com/codeflow?start=8&length=36)

From this it seems like empty string == no negotiated version.  Is that right?

Down below, could we make the catch block return an empty string instead of an exception then?  It seems like the first time will throw, and the second time will return an empty string.  If I'm reading that right, I would prefer they were consistent.

Let me also make a note to go edit our inMimiumBrokerVerisonSupported* function because right now it has this block, which I think would be wrong:
String version = mController.hello(mStrategy, brokerVersion);
                if (version != null) {
                    return true;
                }

Reply:
Nevermind, I see now this is still a throw.  Would it be breaking to swap this to a sentinel empty string value instead? I'm not sure what the general Java opinion is on throwing in 'expected' failure cases like the broker version not being new enough that we'll hit on every MSA request.

Reply:
Thanks for catching. Switched back to isEmpty.

In theory, it can be breaking since for a handshake failure we have always returned this exception.
I kept the same contract with cache also.